envoy.dependency.check/envoy/dependency/check/abstract/dependency.py

```diff
@@ -133,7 +137,7 @@ async def recent_commits(self) -> int:
     @cached_property
     def release(self) -> "abstract.ADependencyGithubRelease":
         """Github release."""
-        return self.release_class(self.repo, self.github_version)
+        return self.release_class(self.repo, self.github_version, session=self.session, url=self.metadata["urls"])
```

Comment:
likewise we can get the `session` from the repo i think - im wondering about the url - perhaps we should just give the relaease classe the metadata object or at least make the url a mandatory arg